django/conf/__init__.py

```diff
@@ -109,6 +109,21 @@ def __init__(self, settings_module):
                 setattr(self, setting, setting_value)
                 self._explicit_settings.add(setting)
 
+        if self.SECRET_KEY_FILE and not self.SECRET_KEY:
+            import stat
+
+            # Verify that file permissions are set to be
+            # only user readable and writable.
+            mode = os.stat(self.SECRET_KEY_FILE).st_mode
+            if bool((stat.S_IRGRP | stat.S_IWGRP | stat.S_IXGRP | stat.S_IROTH | stat.S_IWOTH | stat.S_IXOTH) & mode):
+                raise FilePermissionError("The SECRET_KEY_FILE permissions are not secure.  Set the file permissions to be only user readable and writable.")
```

Comment:
Personally, this would be too strict for my situation. I would need group read and write. Generating the file with user-only permissions would be fine for me.